higan/target-web/shell.html

```diff
@@ -1097,51 +1309,178 @@ <h1>👋First time around?</h1>
         document.body.requestFullscreen();
       }
 
-      // Configure the controller buttons, and make the controller appear on screen
-      function setupAndShowController(controller, port, type) {
-        const buttons = controller.querySelectorAll('button');
+      // Show the right controller(s) based on the selected emulator
+      function displayControllerForEmulator(name) {
+        Module.connectPeripheral(p1Port, 'Gamepad');
+        emulatorName = name;
 
-        controller.display = 'block';
-        controller.style.opacity = 1;
-        controller.style.bottom  = 0;
+        virtualGamepadElement.style.display = 'block';
 
-        // todo: keep track of peripheral and disconnect it on unload
-        Module.connectPeripheral(port, type);
-        buttons.forEach((button) => {
-          button.addEventListener('touchstart', () => Module.setButton(port, button.value, 1));
-          button.addEventListener('touchend', () => Module.setButton(port, button.value, 0));
-        });       
+        // remove existing buttons
+        while (virtualGamepadElement.firstChild) {
```

Comment:
If you look at the current code I believe we were doing a reset by overwriting the HTML content in the container instead, which would be more efficient (and potentially less risky) than processing each button element.

Reply:
ok, updated